src/main/kotlin/com/zepben/evolve/services/network/translator/NetworkCimToProto.kt

```diff
@@ -536,6 +545,20 @@ fun toPb(cim: ConnectivityNode, pb: PBConnectivityNode.Builder): PBConnectivityN
 fun toPb(cim: ConnectivityNodeContainer, pb: PBConnectivityNodeContainer.Builder): PBConnectivityNodeContainer.Builder =
     pb.apply { toPb(cim, psrBuilder) }
 
+fun toPb(cim: Curve, pb: PBCurve.Builder): PBCurve.Builder =
+    pb.apply {
+        cim.data.forEachIndexed { i, data -> addCurveDataBuilder(i).apply { toPb(data, this) } }
+        toPb(cim, ioBuilder)
+    }
+
+fun toPb(cim: CurveData, pb: PBCurveData.Builder): PBCurveData.Builder =
+    pb.apply {
+        xvalue = cim.xValue
+        y1Value = cim.y1Value
+        cim.y2Value?.let { y2Value = it }
```

Comment:
todo: Should be using `UNKNOWN_DOUBLE` as you have with other calls. Need to check why testing has not detected this.

Reply:
changed to UNKNOWN_FLOAT

Reply:
not sure why not detected, is it because in pb it's default to 0 so even if it's not set it will still be 0?
Also this value should be nullable right so should it be set up differently?